Cloud Docs (#776)
grouparoo/www.grouparoo.com: pages/docs/cloud.mdx

---
title: "Grouparoo Cloud Integration"
date: "2021-11-08"
pullQuote: "An easy-to-follow guide on integrating your Code Config repository with Grouparoo Cloud"
---

> We're excited to share our Cloud offering! This guide should cover everything you need to get integrated

Reviewer: ```suggestion
We're excited to share our Cloud offering! This guide covers everything you need to get integrated
```

using your own repository with Grouparoo Cloud.

### Step 1 - Acquiring an API Key

> As of right now you need to make a [Trial Request](/trial) in order to get Grouparoo Cloud access. Our Administrators should get back to you within a day to help get you set up in our system and ready to integrate! After the setup process you should have an API Key and a Project ID which are required for integration.

Reviewer 2: Perhaps update this to reflect that the API Key can be acquired from the Integrations page

also, maybe check with @andyjih on the "SLA" or whatever the right term is for getting back to them (referring to `Our Administrators should get back to you within a day`)

Reviewer: Yeah, +1 to removing the timeline SLA. Just saying signing up for a Trial Request will get you access to the API Key is fine for now.

Reviewer: ```suggestion
To get started, you need to make a [Trial Request](/trial) to get Grouparoo Cloud access. Our Administrators will help get you set up in our system and ready to integrate! After the setup process you will have an API Key and a Project ID which are required for integration.
```

### Step 2 - Integrating your Code Config repository with Grouparoo Cloud

> If you do not have a Code Config repository set up, please follow our [Docs](/docs/getting-started) on how to do so.

Reviewer 2: This doesn't quite line up with the sequence of events- The repository should be created, then the action added, and _then_ we can upload the grouparoo project

Author: This comment is really in here to tell them if they've gotten off-track.

---
title: "Grouparoo Cloud Integration"
date: "2021-11-08"
pullQuote: "An easy-to-follow guide on integrating your Code Config repository with Grouparoo Cloud"
---

We're excited to share our Cloud offering! This guide should cover everything you need to get integrated
using your own repository with Grouparoo Cloud.

### Step 1 - Getting to your Project

Sign in to your [Cloud](http://cloud.grouparoo.com/) account, or create a new one.

#### If a new account

Follow the steps shown on Cloud to create an Organzation and Project.

#### If an existing account, but a new Grouparoo Project

Use the Project dropdown to create a new Project.
screenshot here

#### If an existing account, and project

You're set, follow the steps below!

### Step 2 - Acquring an API Key

Get your API Key. This can be found on your Project's primary page by clicking the Key icon near
the top right of the page.

> API Keys are created with your project! Press the Copy button to copy the API Key.

Reviewer 2: Steps to create a project / apikey would be great to have, but we don't really support it quite yet. I think we should probably only put things in the docs that are available to avoid confusion, and assume that if they are setting up their repo then they already have a project set up.

### Step 3 - Creating your Github action

Note: Until Cloud is up and running, be sure to run the `grouparoo-infra-apps -> cloud-terraformer` Github action with your project's name!

> 

Reviewer 2: This is an internal detail that should not be in the public docs

Author: For sure. This is here so I don't forget during future testing and won't make it into the final version.

The best way to interact with Grouparoo Cloud and Github Actions is as follows:

1. Create a github repo, public or private, to hold your Code Config. Do _not_ push your code config yet!

   - `mkdir my-code-config && cd my-code-config`
   - `git init && git commit -m "first commit" --allow-empty && git remote add origin <my origin.git> && git push -u origin main`
   - Make a new branch for your Code Config, and github action
   - `git checkout -B my-cloud-workflow-branch`
   - `grouparoo init .`
   - Use `grouparoo config` to setup your [Code Config](https://www.grouparoo.com/docs/config). Make sure to commit it!

2. Create the workflow for Grouparoo Cloud and push it up

>    - `mkdir -p .github/workflows && curl https://raw.githubusercontent.com/grouparoo/app-example-cloud/main/.github/workflows/grouparoo-cloud.yml?token=ABF42DR7YI226UKGVOEFI43BRZ346 -o .github/workflows/grouparoo-cloud.yml`

Author: This token was generated by github and gives access to _only_ this file.

   - `git add .github`
   - `git commit -m "Adding the workflow"`

>    - `git push -u origin my-cloud-workflow-branch`

Reviewer 2: We'll probably make `app-example-cloud` public soon to link to, but I think for these docs we should showcase the action contents in a code block and explain a bit about what it's doing (it will run validate on pull requests and will apply the config on merges to `main`) - they can also tweak it to change behavior / branch names

3. Activate the Workflow

>    - Go into your repository's Settings -> Secrets -> New repository secret

Reviewer 2: Could probably add some screenshots here
